zjit/src/codegen.rs

```diff
@@ -67,12 +67,17 @@ impl JITState {
 
 /// CRuby API to compile a given ISEQ
 #[unsafe(no_mangle)]
-pub extern "C" fn rb_zjit_iseq_gen_entry_point(iseq: IseqPtr, _ec: EcPtr) -> *const u8 {
+pub extern "C" fn rb_zjit_iseq_gen_entry_point(iseq: IseqPtr, ec: EcPtr) -> *const u8 {
     // Do not test the JIT code in HIR tests
     if cfg!(test) {
         return std::ptr::null();
     }
 
+    // Check if c_call or c_return tracing is enabled - if so, don't compile
+    if unsafe { rb_c_method_tracing_currently_enabled(ec) } {
```

Comment:
Why check that here? If you're trying to mimic what YJIT does, it should be a side exit on HIR instructions for C method calls (or dynamic dispatch that can be C method calls). We generally don't want to fail the compilation (`return null()`).